src/algebra/big_operators/fin.lean

```diff
@@ -139,6 +142,40 @@ by simpa only [prod_univ_add, fintype.prod_eq_one _ hf, mul_one]
 
 end fin
 
+/-- Equivalence between `fin n → fin m` and `fin (m ^ n)`. -/
+def fin_function_fin_equiv {m n : ℕ} : (fin n → fin m) ≃ fin (m ^ n) :=
+equiv.of_right_inverse_of_card_le
```

Comment:
I'm guessing proving the other direction was really nasty?

Reply:
I just could not finish the proof because it involved rewriting `(∑ i, f i) / a` as `∑ i, f i / a` and this is hardly true for general naturals.

Reply:
It can't because it requires stuff `algebra.big_operators.fin` (and I don't think it makes sense to turn the import around).